pilot/pkg/model/config.go

```diff
@@ -1062,6 +1088,80 @@ func (store *istioConfigStore) EndUserAuthenticationPolicySpecByDestination(inst
 	return out
 }
 
+// Set jwks_uri through openID discovery if it's not set in auth policy.
+func (store *istioConfigStore) SetAuthenticationPolicyJwksURIs(policy *authn.Policy) {
+	if policy == nil {
+		return
+	}
+
+	for _, method := range policy.Peers {
+		switch method.GetParams().(type) {
+		case *authn.PeerAuthenticationMethod_Jwt:
+			policyJwt := method.GetJwt()
+			if err := store.getJwksURI(policyJwt); err != nil {
+				log.Warnf("Failed to get jwks_uri for issuer %q: %v", policyJwt.Issuer, err)
+				continue
+			}
+		}
+	}
+	for _, method := range policy.Origins {
+		policyJwt := method.GetJwt()
+		if err := store.getJwksURI(policyJwt); err != nil {
+			log.Warnf("Failed to get jwks_uri for issuer %q: %v", policyJwt.Issuer, err)
+			continue
+		}
+	}
+}
+
+// Get jwks_uri through openID discovery if it's not set in auth policy, and cache the jwks_uri for furture use.
+func (store *istioConfigStore) getJwksURI(policyJwt *authn.Jwt) error {
```

Comment:
The function name is a bit misleading. I probably make this function to just fetch and return the jwks uri, and move the policy modification part (i.e set policy's jwksURI) to the SetAuthenticationPolicyJwksURIs function above. It would also be easier to have the test for that function by itself.

Reply:
done.